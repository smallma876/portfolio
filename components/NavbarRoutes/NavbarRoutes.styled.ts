import styled from "styled-components";

const Navbar = styled.ul`
  display: flex;
  list-style: none;
  margin: 0;
  padding: 0;
  color: #fff;
  @media (max-width: 768px) {
    flex-direction: column;
  }
`;

const NavbarItem = styled.li`
  padding: 1rem;
  margin: 0;
  height: 3.5rem;
  :hover {
    background: #ecb365;
    color: #04293a;
  }
`;

export const NavbarRoutesStyles = {
  Navbar,
  NavbarItem,
};
