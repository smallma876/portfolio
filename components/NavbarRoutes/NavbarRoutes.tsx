import Link from "next/link";
import React from "react";
import { navbarRouter } from "../../global";
import { Box, Flexbox } from "../../shared/components/Flexbox/Flexbox";
import { useWindowSize } from "../../utils/useWindowResize";
import { NavbarRoutesStyles } from "./NavbarRoutes.styled";

const NavbarRoutes = () => {

  return (
    <NavbarRoutesStyles.Navbar>
      {navbarRouter.map(({ name, path }) => (
        <NavbarRoutesStyles.NavbarItem key={name}>
          <Link href={path}>{name}</Link>
        </NavbarRoutesStyles.NavbarItem>
      ))}
    </NavbarRoutesStyles.Navbar>
  );
};

export default NavbarRoutes;

