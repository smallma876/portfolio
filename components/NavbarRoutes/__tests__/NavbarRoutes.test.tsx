import { render, screen } from "@testing-library/react";
import NavbarRoutes from "../NavbarRoutes";

const renderComponent = async () => {
  const { container } = await render(<NavbarRoutes />);

  return container;
};

describe("NavbarRoutes", () => {
  it("should render match snapshot", () => {
    const container = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
