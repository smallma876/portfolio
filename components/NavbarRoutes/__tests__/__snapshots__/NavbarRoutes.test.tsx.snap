// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarRoutes should render match snapshot 1`] = `Promise {}`;
